gdc/docker-compose.yml

```diff
@@ -92,10 +92,54 @@ services:
     volumes:
       - dynamodb-data:/home/dynamodblocal/data
 
+# ---------- OpenSearch ----------
+  opensearch:
+    image: opensearchproject/opensearch:2.5.0
+    environment:
+      - cluster.name=opensearch-cluster # Name the cluster
+      - node.name=opensearch # Name the node that will run in this container
+      - discovery.seed_hosts=opensearch # Nodes to look for when discovering the cluster
+      - cluster.initial_cluster_manager_nodes=opensearch # Nodes eligibile to serve as cluster manager
+      - bootstrap.memory_lock=true # Disable JVM heap memory swapping
+      - "OPENSEARCH_JAVA_OPTS=-Xms512m -Xmx512m" # Set min and max JVM heap sizes to at least 50% of system RAM
+      - "DISABLE_INSTALL_DEMO_CONFIG=true" # Prevents execution of bundled demo script which installs demo certificates and security configurations to OpenSearch
+      - "DISABLE_SECURITY_PLUGIN=true" # Disables security plugin
+    ulimits:
+      memlock:
+        soft: -1 # Set memlock to unlimited (no soft or hard limit)
+        hard: -1
+      nofile:
+        soft: 65536 # Maximum number of open files for the opensearch user - set to at least 65536
+        hard: 65536
```

Comment:
These seem... kind of dangerous? Is there a reason we're overriding the defaults here?

Reply:
We're doing this since this is what the official documentation has as well: https://opensearch.org/docs/latest/install-and-configure/install-opensearch/docker/#sample-docker-composeyml